example/frontend-testing/rest.js

```diff
@@ -0,0 +1,197 @@
+let selectedFile = null;
+const responseTextContainer = document.getElementById('transcriptionResponse');
+const transcriptionFileName = document.getElementById('transcriptionFileName');
+
+/**
+ * On window load, retrieve and set the API key from localStorage.
+ * If an API key is found, it is populated in the input field.
+ */
+window.onload = function() {
+    // Retrieve the stored API key from localStorage
+    const storedApiKey = localStorage.getItem('apiKey');
+    
+    // If an API key exists, set it in the input field
+    if (storedApiKey) {
+        document.getElementById('apiKey').value = storedApiKey;
+    }
+};
+
+/**
+ * Function to check the health status of the API on window load.
+ * Updates the 'checkHealth' element with the status of the service.
+ */
+window.onload = async function checkHealth() {
+    const responseContainer = document.getElementById('checkHealth');
+    responseContainer.textContent = 'Calling API...';
+
+    const apiKey = localStorage.getItem('apiKey');  // Retrieve the saved API key
+    if (!apiKey) {
+        responseContainer.textContent = 'API key is missing. Please save your API key first.';
+        return;
+    }
+
+    try {
+        // Make a GET request to the health check endpoint
+        const response = await fetch('http://localhost:8393/health', {
+            method: 'GET',
+            headers: {
+                'Authorization': `${apiKey}`
+            }
+        });
+
+        // Check if the response is successful
+        if (!response.ok) {
+            throw new Error(`Error: ${response.status}`);
+        } else {
+            // Update the response container with the service status
+            responseContainer.textContent = 'Authorized - Service is running';
+        }
+    } catch (error) {
```

Comment:
You throw the error to catch it again? Is there another function here that can throw an error? Otherwise I like the idea here but if you just try to catch your one (own) error then this try catch thing seems more confusing than it needs to be (at least to me)